File: leo2-node/src/main/java/org/deku/leo2/node/data/repositories/master/CountryRepository.java
package org.deku.leo2.node.data.repositories.master;

import org.deku.leo2.node.data.entities.master.Country;
import org.deku.leo2.node.data.repositories.master.custom.CountryRepositoryCustom;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QueryDslPredicateExecutor;

/**
 * Created by JT on 11.05.15.
 */
public interface CountryRepository extends JpaRepository<Country, String>, QueryDslPredicateExecutor, CountryRepositoryCustom {
}

File: leoz-node/src/main/java/org/deku/leoz/node/data/repository/master/CountryRepository.kt
package org.deku.leoz.node.data.repository.master

import org.deku.leoz.node.data.jpa.MstCountry
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QueryDslPredicateExecutor
import javax.inject.Inject

/**
 * Country repository
 * Created by masc on 11.05.15.
 */
interface CountryRepository :
        JpaRepository<MstCountry, String>,
        QueryDslPredicateExecutor<MstCountry>,
        CountryRepositoryExtension


interface CountryRepositoryExtension

class CountryRepositoryImpl : CountryRepositoryExtension {
    @Inject
    private lateinit var routeRepository: CountryRepository
}
